crates/bevy_reflect/src/serde/ser.rs

```diff
@@ -39,14 +39,20 @@ fn get_serializable<'a, E: Error>(
         ))
     })?;
 
-    let reflect_serialize = type_registry
-        .get_type_data::<ReflectSerialize>(info.type_id())
-        .ok_or_else(|| {
-            Error::custom(format_args!(
-                "Type '{}' did not register ReflectSerialize",
-                info.type_path(),
-            ))
-        })?;
+    let registration = type_registry.get(info.type_id()).ok_or_else(|| {
+        Error::custom(format_args!(
+            "Type `{}` is not registered in the type registry",
+            info.type_path(),
+        ))
+    })?;
+
+    let reflect_serialize = registration.data::<ReflectSerialize>().ok_or_else(|| {
+        Error::custom(format_args!(
+            "Type `{}` did not register the `ReflectSerialize` type data. For certain types, this may need to be registered manually using `register_type_data`",
```

Comment:
Note: I put `register_type_data` instead of `TypeRegistry::register_type_data` because users will probably more often be going through `App::register_type_data`. However, `App` doesn't exist in `bevy_reflect`, so I chose to keep it vague. If they try looking for it in the docs, then chances are they'll see both.